### PR TITLE
docs: add "Sharing state between islands" example

### LINKED ISSUE #1248: State sharing between islands using signals
Needs a docs page, maybe also a Getting Started entry

### PR DESCRIPTION
**Pull request by deer**
closes https://github.com/denoland/fresh/issues/1248

I'm definitely not sure if this is what Luca was looking for when he opened 1248. I've basically just lifted the code from https://fresh-with-signals.deno.dev/ with a minimal amount of explanation. Breaking it down like this was super helpful for my personal understanding; not sure if it's good for others.

### CODE REVIEW

File: docs/examples/sharing-state-between-islands.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+here, yet.
+
+## Multiple Sibling Islands with Shared State
+
+But we can switch things up by looking at a `Slider.tsx` like this:
+
+```ts
+import { signal } from "@preact/signals";
+
+const slider = signal(50);
```
Comment (marvinhagemeister) [inline docs/examples/sharing-state-between-islands.md]: Whilst this is a valid pattern, let's push folks to use `useSignal` instead. Like a good example would be to instantiate `useSignal` inside a route file and then pass that to two different islands.

The signal here is not just shared between islands, but also for every request, which is likely unintended.

Comment (deer) [reply docs/examples/sharing-state-between-islands.md]: Sadly I don't follow. This is in an island. How is the signal shared across every request?

I can rework this to use `useSignal` and then pass it in instead.

Comment (marvinhagemeister) [reply docs/examples/sharing-state-between-islands.md]: Right, I missed that this is in an island. Still showing how signals are shared outside of that is more useful which means they'll be passed as props to the island.

File: docs/examples/sharing-state-between-islands.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  const sliderSignal = useSignal(50);
+  return (
+    <div>
+      <SynchronizedSlider slider={sliderSignal} />
+      <SynchronizedSlider slider={sliderSignal} />
+      <SynchronizedSlider slider={sliderSignal} />
+    </div>
+  );
+}
+```
```
Comment (marvinhagemeister) [inline docs/examples/sharing-state-between-islands.md:87]: This is great!!

File: flaky_test_tester.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+async function main() {
+  const runs = 100;
+  let failures = 0;
+
+  for (let i = 0; i < runs; i++) {
```
Comment (marvinhagemeister) [inline flaky_test_tester.ts]: Was this file committed accidentally?

Comment (deer) [reply flaky_test_tester.ts]: !!!!! 🙈

Comment (deer) [reply flaky_test_tester.ts]: This is what multitasking does :(

**Comment by deer** [comment]
Is there a way to retrigger tests without a new commit?

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**